File: src/main/java/kernitus/plugin/OldCombatMechanics/utilities/damage/WeaponDamages.java
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package kernitus.plugin.OldCombatMechanics.utilities.damage;

import com.cryptomorin.xseries.XMaterial;
import kernitus.plugin.OldCombatMechanics.OCMMain;
import kernitus.plugin.OldCombatMechanics.utilities.ConfigUtils;
import org.bukkit.Material;
import org.bukkit.configuration.ConfigurationSection;

import java.util.HashMap;
import java.util.Map;

public class WeaponDamages {

    private static Map<String, Double> damages;

    private static OCMMain plugin;

    public static void initialise(OCMMain plugin) {
        WeaponDamages.plugin = plugin;
        reload();
    }

    private static void reload() {
        final ConfigurationSection section = plugin.getConfig().getConfigurationSection("old-tool-damage.damages");
        damages = ConfigUtils.loadDoubleMap(section);
    }

    public static double getDamage(Material mat) {
        final String name = mat.name().replace("GOLDEN", "GOLD").replace("WOODEN", "WOOD").replace("SHOVEL", "SPADE");
        return damages.getOrDefault(name, -1.0);
    }

    public static double getDamage(String key) {
        return damages.getOrDefault(key, -1.0);
    }

    public static Map<Material, Double> getMaterialDamages() {
        final Map<Material, Double> materialMap = new HashMap<>();
        damages.forEach((name, damage) -> {
            final String newName = name.replace("GOLD", "GOLDEN").replace("WOOD", "WOODEN").replace("SPADE", "SHOVEL");
            XMaterial.matchXMaterial(newName).ifPresent(xmaterial -> {
                final Material material = xmaterial.parseMaterial();
                if (material != null) {
                    materialMap.put(material, damage);
                }
            });
        });
        return materialMap;
    }
}

File: src/main/kotlin/kernitus/plugin/OldCombatMechanics/utilities/damage/WeaponDamages.kt
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package kernitus.plugin.OldCombatMechanics.utilities.damage

import kernitus.plugin.OldCombatMechanics.OCMMain
import kernitus.plugin.OldCombatMechanics.utilities.ConfigUtils
import org.bukkit.Material
import java.util.*

object WeaponDamages {
    private var damages: Map<String, Double> = emptyMap()

    private lateinit var plugin: OCMMain

    fun initialise(plugin: OCMMain) {
        WeaponDamages.plugin = plugin
        reload()
    }

    private fun reload() {
        val section = plugin.config.getConfigurationSection("old-tool-damage.damages")
        damages = ConfigUtils.loadDoubleMap(section!!) // Should error out if damages not found
    }

    fun getDamage(mat: Material): Double {
        val name = mat.name.replace("GOLDEN", "GOLD").replace("WOODEN", "WOOD").replace("SHOVEL", "SPADE")
        return damages.getOrDefault(name, -1.0)
    }

    val materialDamages: Map<Material, Double>
        get() {
            val materialMap: MutableMap<Material, Double> =
                EnumMap(org.bukkit.Material::class.java)
            damages.forEach { (name: String, damage: Double) ->
                val newName =
                    name.replace("GOLD", "GOLDEN").replace("WOOD", "WOODEN").replace("SPADE", "SHOVEL")
                materialMap[Material.valueOf(newName)] = damage
            }
            return materialMap
        }
}
